### PR TITLE
[python-experimental] Enhance octet-stream deserialize

### PR DESCRIPTION
**Pull request by EltonChou**
@spacether 
+ Add a method to extract filename when the headers of octet-stream resopnse didn't provide the filename.
+ Add test for the method.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: samples/openapi3/client/petstore/python-experimental/tests_manual/test_fake_api.py [hunk 1]
```diff
@@ -400,6 +400,38 @@ def close(self):
...
         os.unlink(api_response.body.name)
 
+        file1 = open(file_path1, "rb")
+        streamable_body = StreamableBody(file1)
+        expected_filename = "the_file.ext"
+
+        no_filename_mock_response = self.response(
+            streamable_body,
+            content_type='application/octet-stream',
+            headers={'content-disposition': f'attachment'},
```
Comment (spacether) [inline samples/openapi3/client/petstore/python-experimental/tests_manual/test_fake_api.py:410]: nit: one does not need a f-string for the content disposition value. How about removing the f prefix from that string?

**Review by spacether — CHANGES_REQUESTED** [review]
Thank you for your PR. In general it looks great. Can you tweak your __file_name_from_response_url to return early when None is passed into it? That will make it quicker.

File: samples/openapi3/client/petstore/python-experimental/petstore_api/api_client.py [hunk 2]
```diff
@@ -836,6 +836,17 @@ def __deserialize_json(response: urllib3.HTTPResponse) -> typing.Any:
         # python must be >= 3.9 so we can pass in bytes into json.loads
         return json.loads(response.data)
 
+    @staticmethod
+    def __file_name_from_response_url(response_url: typing.Optional[str]) -> typing.Optional[str]:
+        url_path = urlparse(response_url).path
```
[the remark this answers is not in the document]
Comment (EltonChou) [reply samples/openapi3/client/petstore/python-experimental/petstore_api/api_client.py:841]: Sure. I will push another commit to fix this.

**Review by spacether — CHANGES_REQUESTED** [review]
Thank you for making those updates.
Your PR broke the 3_0_3_unit_test tests in CI on node 4 when you removed these files:
- samples/openapi3/client/3_0_3_unit_test/python-experimental/test/__init__.py
- samples/openapi3/client/3_0_3_unit_test/python-experimental/test/test_models/__init__.py

Can you please add those files back in?

**Comment by EltonChou** [comment]
> Thank you for making those updates. Your PR broke the 3_0_3_unit_test tests in CI on node 4 when you removed these files:
> 
> * samples/openapi3/client/3_0_3_unit_test/python-experimental/test/**init**.py
> * samples/openapi3/client/3_0_3_unit_test/python-experimental/test/test_models/**init**.py
> 
> Can you please add those files back in?

Thanks for your review.
I didn't notice the generating script was terminated accidentally.
I updated the samples.

**Review by spacether — APPROVED** [review]
Your PR looks great, thank you for the feature addition!

**Pull request merged**